src/openrct2/ride/Vehicle.cpp

```diff
@@ -6246,14 +6248,17 @@ void Vehicle::UpdateAdditionalAnimation()
 static void play_scenery_door_open_sound(const CoordsXYZ& loc, WallElement* tileElement)
 {
     auto* wallEntry = tileElement->GetEntry();
-    int32_t doorSoundType = WallEntryGetDoorSound(wallEntry);
-    if (doorSoundType != 0)
+    if (wallEntry == nullptr)
+        return;
+
+    auto doorSoundType = wallEntry->getDoorSoundType();
+    if (doorSoundType == DoorSoundType::none)
+        return;
+
+    auto soundId = DoorOpenSoundIds[EnumValue(doorSoundType)];
+    if (soundId != OpenRCT2::Audio::SoundId::Null)
```

Comment:
Is this check even required? We know that none maps to null and we check for none already.

Reply:
Strictly speaking, it isn’t required. Shall I remove it?

Reply:
Its a condition that can never happen so I think its best to remove it.